saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java

```diff
@@ -299,6 +299,33 @@ public static InputStream getSourceAsStream(Source source) throws ConfigurationE
         return new ByteArrayInputStream(baos.toByteArray());
     }
 
+    /**
+     * Stream a DOM Node as a String
+     *
+     * @param node
+     *
+     * @return
+     *
+     * @throws ProcessingException
+     * @throws TransformerFactoryConfigurationError
+     * @throws TransformerException
+     */
+    public static String getNodeAsString(Node node) throws ConfigurationException, ProcessingException {
```

Comment:
I just noticed that this method org.keycloak.saml.common.util.DocumentUtil#getDocumentAsString does the same thing.

Does it work for you?

Reply:
@pedroigor 
We can't call `DocumentUtil#getDocumentAsString` directly because of the type miss match of the argument. How about refactoring the `getDocumentAsString` method and adding `getNodeAsString` method as follows?

```
    public static String getDocumentAsString(Document signedDoc) throws ProcessingException, ConfigurationException {
        return getNodeAsString(signedDoc);
    }

    public static String getNodeAsString(Node node) throws ConfigurationException, ProcessingException {
        Source source = new DOMSource(node);
        StringWriter sw = new StringWriter();

        Result streamResult = new StreamResult(sw);
        // Write the DOM document to the stream
        Transformer xformer = TransformerUtil.getTransformer();
        try {
            xformer.transform(source, streamResult);
        } catch (TransformerException e) {
            throw logger.processingError(e);
        }

        return sw.toString();
    }
```

Reply:
It works. I think that is what I was trying to say but I forgot you are actually using a Node and not a Document.